src/core/src/package_managers/TdnfPackageManager.py

```diff
@@ -0,0 +1,458 @@
+
+    def get_package_version_without_epoch(self, package_version):
+        """Returns the package version stripped of any epoch"""
+        # todo: verify usage
+        package_version_split = str(package_version).split(':', 1)
+
+        if len(package_version_split) == 2:
+            self.composite_logger.log_verbose("[TDNF]   > Removed epoch from version (" + package_version + "): " + package_version_split[1])
+            return package_version_split[1]
+
+        if len(package_version_split) != 1:
+            self.composite_logger.log_error("[TDNF] Unexpected error during version epoch removal from package version. [PackageVersion={0}]".format(package_version))
+
+        return package_version
+
+    def get_package_size(self, output):
+        """Retrieve package size from installation output string"""
+        # Sample output line:
+        # Total download size: 15 M
+        if "Nothing to do" not in output:
+            lines = output.strip().split('\n')
+            for line in lines:
+                if line.find(self.STR_TOTAL_DOWNLOAD_SIZE) >= 0:
+                    return line.replace(self.STR_TOTAL_DOWNLOAD_SIZE, "")
+
+        return Constants.UNKNOWN_PACKAGE_SIZE
+    # endregion
+
+    # region auto OS updates
+    # endregion
+
+    # region Reboot Management
+    def is_reboot_pending(self):
+        """ Checks if there is a pending reboot on the machine. """
+        try:
+            pending_file_exists = os.path.isfile(self.REBOOT_PENDING_FILE_PATH)
+            pending_processes_exist = self.do_processes_require_restart()
+            self.composite_logger.log_debug("[TDNF] > Reboot required debug flags (tdnf): " + str(pending_file_exists) + ", " + str(pending_processes_exist) + ".")
+            return pending_file_exists or pending_processes_exist
+        except Exception as error:
+            self.composite_logger.log_error('[TDNF] Error while checking for reboot pending (tdnf): ' + repr(error))
+            return True  # defaults for safety
+
+    def do_processes_require_restart(self):
+        """Signals whether processes require a restart due to updates"""
+        self.composite_logger.log_verbose("[TDNF] Checking if process requires reboot")
+        # Checking using dnf-utils
+        code, out = self.env_layer.run_command_output(self.dnf_utils_prerequisite, False, False)  # idempotent, doesn't install if already present
+        self.composite_logger.log_verbose("[TDNF] Idempotent dnf-utils existence check. [Code={0}][Out={1}]".format(str(code), out))
+
+        # Checking for restart for distros with -r flag
+        code, out = self.env_layer.run_command_output(self.needs_restarting_with_flag, False, False)
+        self.composite_logger.log_verbose("[TDNF] > Code: " + str(code) + ", Output: \n|\t" + "\n|\t".join(out.splitlines()))
+        if out.find("Reboot is required") < 0:
```

Comment:
Please share full output of the command in a reboot-required scenario (PR comment).
This will effectively be covered in UTs too, but good to have an early check.

Reply:
Covered in the log 2.core.log from 'With package exclusions' 
<img width="802" alt="image" src="https://github.com/user-attachments/assets/90da46b2-2664-4c92-8134-8a84b10c1224" />